### DISCUSSION TITLE
どこかでショートしているが、デバッグ方法がわかりません

### DISCUSSION THREAD

**Discussion by ghost** [Yowkees/keyball#338] [Q&A]
![left](https://github.com/Yowkees/keyball/assets/144687922/0a5d43c4-f076-4cf3-a5a3-a859f0223951)
■ 左側（トラックボールなし側）にて下記状態です。
・単体だとOLEDが消灯したまま反応しない
・TSSRケーブルを接続して右側にUSBを接続すると、OLEDが表示される（左はKeyballロゴ）
・上図の黄色丸部の4キーが反応しない
・②のキーを押すと、USBがPCから切断されて、ProMicroがフリーズする。
　※右OLEDで確認できていたキー入力表示が反応しなくなる。（表示自体は消えず、②を押した直前の状態のまま）
・上図の黄色丸、オレンジ丸以外については入力可能だが、Layer3で設定されている
キーが入力されてしまう。（おそらく①がショートしたままになっている）
・LEDは全て点灯

![image](https://github.com/Yowkees/keyball/assets/144687922/20e8fedc-a517-4e73-8954-dfc7c275435e)
上図の左から三番目については、ショートしている疑いを持ったため、はみ出ていたハンダを吸い取り線でなるべく取り除き、テスターで左下のホールとショートしていないことを確認しました。

![image](https://github.com/Yowkees/keyball/assets/144687922/19471f46-58f5-4b37-b9dc-64ffa5cd261a)

■右側（上図）にて、下記状態です。
・LEDが１つも点灯しない
・上記以外についてはトラボやOLEDも含め全て正常に動作
・左右のProMicroを入れ替えても、症状変わらず（左右とも上記全ての症状が再現します）。

大変恐縮ですが、よろしくお願いいたします。

以前、Keyball 39を組み立てた際にもこちらでお世話になりました。
39の方は、LED点灯も含め、全て無事正常に動くようになっており、
毎日快適です。ありがとうございます。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ひとまず左手側の症状が致命的ですので、そちらから対処したいと思います。（おそらくProMicroの信号線がGNDとショートしており、ProMicroに負担がかかっていそうです）

問題のある最下段のキーソケット７個ですが、端子に盛られた半田が多いようです。これを適正な量（ソケットの金属端子からはみ出さない量）に、半田吸い取り線等で除去してください。
キーソケットの基板パッドの多くはベタGNDで囲まれています。ベタGNDは白色のレジストに覆われていて半田が付きませんが、キズがつくと半田が付着し、ショートすることがあります（過去に数例、この原因でキーが入力されつづける不具合を確認しています）。

ということで、周辺にショートしているものがないか確認のために半田量を減らすご対応をお願いします。

お手数ですがよろしくお願いいたします。
